atum/src/main/scala/za/co/absa/atum/AtumImplicits.scala

```diff
@@ -260,6 +261,40 @@ trait AtumImplicitsBase {
       dataset
     }
 
+    /**
+     * The method returns ControlMeasure object from the Atum context
+     * @return - ControlMeasure object
+     */
+    def getControlMeasure: ControlMeasure = {
+      atum.preventNotInitialized()
+      atum.controlFrameworkState.accumulator.getControlMeasure
+    }
+
+    /**
+     * The method returns AditionalInfo object from the Atum context
+     * @return - AditionalInfo object
+     */
+    def getAllAdditionalInfo: Map[String, String] = {
+      atum.preventNotInitialized()
+      atum.controlFrameworkState.accumulator.getControlMeasure.metadata.additionalInfo
+    }
+
+    /**
+     * The method returns Option of AditionalInfo value for given key from the Atum context
+     * @param key - the AditionalInfo key
+     * @return - Option[String] with value for given AditionalInfo key
+     */
+    def getAdditionalInfo(key: String): Option[String] = {
+      atum.preventNotInitialized()
+      if (atum.controlFrameworkState.accumulator.getControlMeasure.metadata.additionalInfo.contains(key)) {
+        Option(atum.controlFrameworkState.accumulator.getControlMeasure.metadata.additionalInfo(key))
+      }
+      else {
+        None
+      }
```

Comment:
"Just to fix the code, line 290 should be probably wrapped by `Some`, not `Option`, but the whole thing can be simplified to:
```suggestion
      atum.controlFrameworkState.accumulator.getControlMeasure.metadata.additionalInfo.get(key) // returns Option[V], here Option[String]
```

Reply:
But here, as well as above, I question the purpose. Not your fault particularly, I can see it was in the request. I will challenge it there.